code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm

```diff
@@ -132,12 +132,13 @@
 	use_plasma(200)
 	icon_state = "Defiler Power Up"
 
-	if(!do_after(src, DEFILER_GAS_CHANNEL_TIME, TRUE, 5, BUSY_ICON_HOSTILE))
-		smoke_system = new /datum/effect_system/smoke_spread/xeno/neuro()
-		smoke_system.set_up(1, get_turf(src))
-		to_chat(src, "<span class='xenodanger'>You abort emitting neurogas, your expended plasma resulting in only a feeble wisp.</span>")
-		emitting_gas = FALSE
-		icon_state = "Defiler Running"
+	if(!do_after(src, DEFILER_GAS_CHANNEL_TIME, TRUE, null, USER_ICON_HOSTILE))
+		if(!QDELETED(src))

```

Comment:
Why not put the src above instead of the null?

Reply:
Refraining from double checking src. Also line 136 check is to avoid null. runtimes.